### PR TITLE
Integrate backend APIs into conversation-related functionality

### LINKED ISSUE #30: Integrate backend into Conversations-related functionality
Need to update the relevant state-management logic to use the new state model, like when sending the token at the start of a new WebSocket connection.

### PR DESCRIPTION
**Pull request by Thao-Tran**
Resolves #30.
The important implementations are in the `conversations` store module, the `conversations` and `websocket` services, and all of the models under `src/services/models`.

The `conversations` store module handles everything related to conversations. It tracks the state of the user's conversations and the open conversation. The module also handles how clients send their own Edit and Caret Update messages.

The `ConversationList` component now uses the vuex modules to get all of the user's information and conversations.

The functionality that was in the original `handlers` file for the `Conversations` component folder is now split up between all of the models and in the `conversations` store module. Essentially each type of message is split into its own class and these message classes have a `fromJSON` function to parse a JSON string into an instance of the class and a `handle` function to handle a received message of that type.

### CODE REVIEW

File: src/services/models/userLeave.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  * Parameters:
+  *   userId: int
+  */
+  constructor(userId) {
+    const data = { userId };
+    super(WSMessage.types.UserLeave, data);
+  }
+
+  /**
+  * handle adds a new active user to the state specified by the UserJoin.
```
Comment (igorveselinovic) [inline src/services/models/userLeave.js:15]:
```suggestion
  * handle removes an active user from the state specified by the UserLeave.
```

File: src/services/models/wsmessage.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    requiredProps.forEach((prop) => {
+      if (!(prop in data)) {
+        missingProps.push(prop);
+      }
+    });
+    return missingProps;
+  }
+
+  /**
+  * fromJSON parse JSON string into a WSMessage instance.
```
Comment (igorveselinovic) [inline src/services/models/wsmessage.js:87]:
```suggestion
  * fromJSON parses JSON string into a WSMessage instance.
```

File: src/services/utils.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    const error = text || response.statusText;
+    return Promise.reject(error);
+  }
+  return JSON.parse(text);
+});
+
+/*
+  Returns the Authorization header with the JWT token
+*/
```
Comment (igorveselinovic) [inline src/services/utils.js:22]:
```suggestion
/**
 * Returns the Authorization header with the JWT token
 */
```

File: src/components/conversation/handlers.js [hunk 4]
```diff
@@ -560,9 +560,9 @@ function parseWSMessage(e) {
   }
 }
 
-function connectWebSocket() {
+function connectWebSocket(backend) {
```
Comment (igorveselinovic) [inline src/components/conversation/handlers.js:563]: I think this whole file can maybe just be removed?

Comment (Thao-Tran) [reply src/components/conversation/handlers.js:563]: Yeah pretty much everything in that folder except for the component can be removed.

**Pull request merged**